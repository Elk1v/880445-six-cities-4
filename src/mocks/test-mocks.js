import moment from "moment";

export const Options = {
  OFFERS_COUNT: 312,
};

export const offers = [
  {
    id: 0,
    title: `Beautiful & luxurious apartment at great location`,
    features: {
      type: `Apartment`,
      bedrooms: 3,
      adults: 4,
    },
    rating: 4.5,
    isPremium: true,
    price: 100,
    images: [
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`
    ],
    inside: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
    ],
    host: {
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`,
      text: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      ]
    },

    coords: [52.3909553943508, 4.85309666406198],

    reviews: [{
      id: 0,
      user: {
        name: `Max`,
        avatar: `img/avatar-max.jpg`
      },
      rating: 4.5,
      text: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      ],
      date: moment().format(`YYYY-MM-DD`),
    }],
  },

  {
    id: 1,
    title: `Beautiful & luxurious apartment at great location`,
    features: {
      type: `Apartment`,
      bedrooms: 3,
      adults: 4,
    },
    rating: 4.0,
    isPremium: false,
    price: 70,
    images: [
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`
    ],
    inside: [
      `Wi-Fi`,
      `Towels`,
      `Coffee machine`,
      `Kitchen`,
      `Dishwasher`,
      `Fridge`,
    ],
    host: {
      name: `Max`,
      avatar: `img/avatar-max.jpg`,
      text: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      ]
    },

    coords: [52.369553943508, 4.85309666406198],

    reviews: [{
      id: 1,
      user: {
        name: `Max`,
        avatar: `img/avatar-max.jpg`
      },
      rating: 4.5,
      text: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      ],
      date: moment().format(`YYYY-MM-DD`),
    }],
  },

  {
    id: 2,
    title: `Beautiful & luxurious apartment at great location`,
    features: {
      type: `Apartment`,
      bedrooms: 3,
      adults: 4,
    },
    rating: 4.5,
    isPremium: true,
    price: 100,
    images: [
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`
    ],
    inside: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
    ],
    host: {
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`,
      text: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      ]
    },

    coords: [52.3909553943508, 4.929309666406198],

    reviews: [{
      id: 2,
      user: {
        name: `Max`,
        avatar: `img/avatar-max.jpg`
      },
      rating: 4.5,
      text: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      ],
      date: moment().format(`YYYY-MM-DD`),
    }],
  },

  {
    id: 3,
    title: `Beautiful & luxurious apartment at great location`,
    features: {
      type: `Apartment`,
      bedrooms: 3,
      adults: 4,
    },
    rating: 4.5,
    isPremium: true,
    price: 100,
    images: [
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`
    ],
    inside: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
    ],
    host: {
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`,
      text: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      ]
    },

    coords: [52.3809553943508, 4.939309666406198],

    reviews: [{
      id: 3,
      user: {
        name: `Max`,
        avatar: `img/avatar-max.jpg`
      },
      rating: 4.5,
      text: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      ],
      date: moment().format(`YYYY-MM-DD`),
    }],
  },

];

export const offer = {
  id: 0,
  title: `Beautiful & luxurious apartment at great location`,
  features: {
    type: `Apartment`,
    bedrooms: 3,
    adults: 4,
  },
  rating: 4.5,
  isPremium: true,
  price: 100,
  images: [
    `img/studio-01.jpg`,
    `img/apartment-01.jpg`,
    `img/apartment-02.jpg`,
    `img/apartment-03.jpg`,
    `img/studio-01.jpg`,
    `img/apartment-01.jpg`
  ],
  inside: [
    `Wi-Fi`,
    `Washing machine`,
    `Towels`,
    `Heating`,
    `Coffee machine`,
    `Baby seat`,
    `Kitchen`,
    `Dishwasher`,
    `Cabel TV`,
    `Fridge`,
  ],
  host: {
    name: `Angelina`,
    avatar: `img/avatar-angelina.jpg`,
    text: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    ]
  },

  coords: [52.3809553943508, 4.939309666406198],

  reviews: {
    id: 0,
    user: {
      name: `Max`,
      avatar: `img/avatar-max.jpg`
    },
    rating: 4.5,
    text: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    ],
    date: moment().format(`YYYY-MM-DD`),
  },
};
