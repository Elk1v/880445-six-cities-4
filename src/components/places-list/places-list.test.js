import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import PlacesList from "PlacesList/places-list";
import {offers} from "Mocks/test-mocks";

const mockStore = configureStore([]);

it(`Should PlaceList component render correcrly`, () => {
  const store = mockStore({
    currentCardId: 0,
  });

  const tree = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <PlacesList
            offers={offers}
            onCardTitleClick={() => {}}
          />
        </Provider>
      </BrowserRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
