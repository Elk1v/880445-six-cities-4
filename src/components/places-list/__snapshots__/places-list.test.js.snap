// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should PlaceList component render correcrly 1`] = `
Array [
  <article
    className="cities__place-card place-card"
    data-key={0}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height={200}
          src="img/studio-01.jpg"
          width={260}
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            100
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button false button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height={19}
            width={18}
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "90%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/property"
          onClick={[Function]}
        >
          Beautiful & luxurious apartment at great location
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>,
  <article
    className="cities__place-card place-card"
    data-key={1}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height={200}
          src="img/apartment-01.jpg"
          width={260}
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            70
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button place-card__bookmark-button--active button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height={19}
            width={18}
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/property"
          onClick={[Function]}
        >
          Beautiful & luxurious apartment at great location
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>,
  <article
    className="cities__place-card place-card"
    data-key={2}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height={200}
          src="img/apartment-02.jpg"
          width={260}
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            100
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button false button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height={19}
            width={18}
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "90%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/property"
          onClick={[Function]}
        >
          Beautiful & luxurious apartment at great location
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>,
  <article
    className="cities__place-card place-card"
    data-key={3}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height={200}
          src="img/apartment-03.jpg"
          width={260}
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            100
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button false button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height={19}
            width={18}
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "90%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/property"
          onClick={[Function]}
        >
          Beautiful & luxurious apartment at great location
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>,
]
`;
